src/traces/parcoords/defaults.js

```diff
@@ -51,29 +51,41 @@ function dimensionsDefaults(traceIn, traceOut) {
             continue;
         }
 
-        coerce('label');
-        coerce('tickvals');
-        coerce('ticktext');
-        coerce('visible');
-        coerce('range');
-        coerce('constraintrange');
-        coerce('values');
+        var userVisible = coerce('visible');
 
         // turn dimensions with no data invisible
-        dimensionOut.visible = dimensionOut.visible && dimensionOut.values.length > 0;
+        var actuallyVisible = userVisible && Lib.isArray(dimensionIn.values) && dimensionIn.values.length > 0;
+
+        if(actuallyVisible) {
```

Comment:
.. just took another look and the above Commit c6b8d34 also sets the `_index`property (besides the above mentioned `values` attribute) whether or not the `dimension` is actually visible. So maybe I misunderstood the part about leaving those `dimension`s alone  (not altering them except their `visible` attribute) that e.g. don't have a valid `values` array, or its length is zero.